src/org/parosproxy/paros/extension/option/OptionsLocalProxyPanel.java

```diff
@@ -1,606 +0,0 @@
-     *
-     * @return ZapPortNumberSpinner
-     */
-    private ZapPortNumberSpinner getSpinnerProxyPort() {
-        if (spinnerProxyPort == null) {
-            // ZAP: Do not allow invalid port numbers
-            spinnerProxyPort = new ZapPortNumberSpinner(8080);
-        }
-        return spinnerProxyPort;
-    }
-
-    /**
-     * This method initializes spinnerReverseProxyHttpPort
-     *
-     * @return ZapPortNumberSpinner
-     */
-    private ZapPortNumberSpinner getSpinnerReverseProxyHttpPort() {
-        if (spinnerReverseProxyHttpPort == null) {
-            // ZAP: Do not allow invalid port numbers
-            spinnerReverseProxyHttpPort = new ZapPortNumberSpinner(80);
-        }
-        return spinnerReverseProxyHttpPort;
-    }
-
-    /**
-     * This method initializes this
-     */
-    private void initialize() {
-        this.setLayout(new CardLayout());
-        this.setName(Constant.messages.getString("options.proxy.local.title"));
-        if (Model.getSingleton().getOptionsParam().getViewParam().getWmUiHandlingOption() == 0) {
-            this.setSize(391, 320);
-        }
-        
-        securityProtocolsPanel = new SecurityProtocolsPanel();
-        this.add(getPanelProxy(), getPanelProxy().getName());
-
-        getChkReverseProxy().setVisible(Constant.isSP());
-        getPanelReverseProxy().setVisible(Constant.isSP());
-
-    }
-
-    @Override
-    public void initParam(Object obj) {
-        OptionsParam optionsParam = (OptionsParam) obj;
-        ProxyParam proxyParam = optionsParam.getProxyParam();
-
-        // set Local Proxy parameters
-        // ZAP: in the Options dialog we can show the real value of the field
-        // and null means that the listener should be bound to all interfaces
-        txtProxyIp.setText(proxyParam.getRawProxyIP());
-        txtProxyIp.discardAllEdits();
-        
-        // ZAP: Do not allow invalid port numbers
-        spinnerProxyPort.setValue(proxyParam.getProxyPort());
-
-        chkBehindNat.setSelected(proxyParam.isBehindNat());
-        chkRemoveUnsupportedEncodings.setSelected(proxyParam.isRemoveUnsupportedEncodings());
-        chkAlwaysDecodeGzip.setSelected(proxyParam.isAlwaysDecodeGzip());
-
-        // set reverse proxy param
-        txtReverseProxyIp.setText(proxyParam.getReverseProxyIp());
-        txtReverseProxyIp.discardAllEdits();
-        
-        // ZAP: Do not allow invalid port numbers
-        spinnerReverseProxyHttpPort.setValue(proxyParam.getReverseProxyHttpPort());
-        spinnerReverseProxyHttpsPort.setValue(proxyParam.getReverseProxyHttpsPort());
-
-        chkReverseProxy.setSelected(proxyParam.isUseReverseProxy());
-        setReverseProxyEnabled(proxyParam.isUseReverseProxy());
-
-        securityProtocolsPanel.setSecurityProtocolsEnabled(proxyParam.getSecurityProtocolsEnabled());
-    }
-
-    @Override
-    public void validateParam(Object obj) throws Exception {
-        securityProtocolsPanel.validateSecurityProtocols();

```

Comment:
Just noticed that this validation was not migrated into the `OptionsProxiesPanel`.